chia/consensus/blockchain.py

```diff
@@ -338,78 +337,18 @@ async def add_block(
-                # for it. This can potentially be quite expensive and we should
-                # try to avoid getting here
-
-                # first, collect all the block hashes of the forked chain
-                # the block we're trying to add doesn't exist in the chain yet,
-                # so we need to start traversing from its prev_header_hash
-                fork_chain, fork_hash = await lookup_fork_chain(
-                    self,
-                    (peak.height, peak.header_hash),
-                    (block.height - 1, block.prev_header_hash),
-                    self.constants,
-                )
-                # now we know how long the fork is, and can compute the fork
-                # height.
-                fork_height = block.height - len(fork_chain) - 1
-                fork_info = ForkInfo(fork_height, fork_height, fork_hash)
-
-                log.warning(
-                    f"slow path in block validation. Building coin set for fork ({fork_height}, {block.height})"
-                )
-
-                # now run all the blocks of the fork to compute the additions
-                # and removals. They are recorded in the fork_info object
-                counter = 0
-                start = time.monotonic()
-                for height in range(fork_info.fork_height + 1, block.height):
-                    fork_block: Optional[FullBlock] = await self.block_store.get_full_block(fork_chain[uint32(height)])
-                    assert fork_block is not None
-                    assert fork_block.height - 1 == fork_info.peak_height
-                    assert fork_block.height == 0 or fork_block.prev_header_hash == fork_info.peak_hash
-                    await self.run_single_block(fork_block, fork_info)
-                    counter += 1
-                end = time.monotonic()
-                log.info(
-                    f"executed {counter} block generators in {end - start:2f} s. "
-                    f"{len(fork_info.additions_since_fork)} additions, "
-                    f"{len(fork_info.removals_since_fork)} removals"
-                )
+        if extending_main_chain:
+            fork_info.reset(block.height - 1, block.prev_header_hash)
 
-        else:
+        block_rec = await self.get_block_record_from_db(header_hash)
+        if block_rec is not None:
             if extending_main_chain:
-                fork_info.reset(block.height - 1, block.prev_header_hash)
-
-            block_rec = await self.get_block_record_from_db(header_hash)
-            if block_rec is not None:
-                # We have already validated the block, but if it's not part of the
-                # main chain, we still need to re-run it to update the additions and
-                # removals in fork_info.
                 await self.advance_fork_info(block, fork_info)
                 fork_info.include_spends(pre_validation_result.conds, block, header_hash)
-                self.add_block_record(block_rec)
```

Comment:
are you sure we don't need this? If this happens because of a deep reorg, `block_rec` may have been pulled from the disk, and may not be in the block cache.
If we're already guaranteed that it's in the cache, can we change the `get_block_record_from_db()` to just `try_block_record()` instead?

Reply:
it looked a littel redundant to me but maybe your right, ill revert this